owners/src/parser.js

```diff
@@ -61,19 +206,29 @@ class OwnersParser {
 
     if (lines instanceof Array) {
       const stringLines = lines.filter(line => typeof line === 'string');
```

Comment:
Did you consider doing the iterations here in the higher level function and using a switch decide which parser method to call? If you do something like that you can remove the `.forEach`'s in the sub-methods and make them only parse a single "line" at a time

```javascript
for (const line of lines) {
  switch (typeof line) {
    case 'string':
      …
      break;

    case 'object':
      …
      break;
    }
  }
}
```

Reply:
That doesn't really work out the way I'd like, since all string lines should be in the owners list for a single rule (not N separate single-owner rules). Meanwhile, each dict line needs to be parsed separately. Added to that, this approach lets me re-use `parseOwnersList` between top-level and dict-level owners lists. I'm not sure what `forEach`'s you're talking about removing; but I did try a handful of ways of shifting things around, and this came out to be among the most sane